jscomp/others/js_promise2.ml

```diff
@@ -0,0 +1,16 @@
+include Js_promise
+
+let then_ : 'a promise -> ('a -> 'b promise) -> 'b promise =
+  [%raw {|
+  function(p, cont) {
+    Promise.resolve(p).then(cont)
```

Comment:
If p is promise, isn't just `p.then(cont)` enough?

Reply:
That's the type unsafe version.
With `resolve`, it is type safe.

Reply:
For a longes answer, see https://github.com/rescript-lang/rescript-compiler/issues/5707